src/catalog/CatalogPage.jsx

```diff
@@ -0,0 +1,95 @@
+import messages from './messages';
+
+const buildLmsUrl = (absoluteUrl) => `${getConfig().LMS_BASE_URL}${absoluteUrl}`;
+const buildCourseURL = (courseKey) => buildLmsUrl(`/courses/${courseKey}/about`);
+
+export const CatalogPageBase = ({
+  intl, fetching, errors, courses, fetchCourses,
+}) => {
+  // This is a React hook. By providing no dependencies in the second
+  // argument, we signal that this hook should be run when the
+  // component is mounted.
+  //
+  // Once the data is updated by the hook, the component will
+  // auto-update as well.
+  useEffect(() => {
+    fetchCourses();
+  }, []);
+  return (
+    <Container>
+      <Row>
+        <Col xs={12}>
+          <h1>{intl.formatMessage(messages['catalog.heading'])}</h1>
+        </Col>
+        <Col xs={12}>
+          {errors.map((error) => <Alert type="danger" key={error}>{error}</Alert>)}
+        </Col>
+      </Row>
+      {(fetching && (
+        <Row>
+          <Col className="text-center">{intl.formatMessage(messages['catalog.loading'])}</Col>
+        </Row>
+      )) || (
+        <Row>
+          {courses.map((course) => (
+            <Col xs={6} md={4} lg={3} key={course.id}>
+              <Card>
+                <Card.Img
+                  variant="top"
+                  src={buildLmsUrl(course.media.course_image.uri)}
+                  alt={`Showcase image for ${course.name}`}
+                />
+                <Card.Header title={course.name} />
+                {course.short_description && <Card.Text>{course.short_description}</Card.Text>}
+                <Button variant="primary" href={buildCourseURL(course.id)}>{intl.formatMessage(messages['catalog.course.view'])}</Button>
+              </Card>
+            </Col>
+          ))}
+        </Row>
+      )}
+    </Container>
+  );
+};
+
+// This is a simplified shape based on the properties we expect to need.
+// If you're running the devstack locally, you can see the full structure
+// by visiting http://localhost:18000/api/courses/v1/courses/
+const CourseShape = PropTypes.shape({
+  media: PropTypes.shape({
+    course_image: PropTypes.shape({
+      uri: PropTypes.string.isRequired,
+    }),
+  }),
+  short_description: PropTypes.string,
+  name: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
+});
+
+CatalogPageBase.propTypes = {
+  intl: intlShape.isRequired,
+  fetching: PropTypes.bool.isRequired,
+  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
+  courses: PropTypes.arrayOf(CourseShape).isRequired,
+  fetchCourses: PropTypes.func.isRequired,
+};
+
+export const CatalogPage = connect(
+  // The selector will return an object whose keys are fed as arguments to our component.
+  selectCourses,
+  // These functions will be bound within redux so they are called with the relevant dispatchers, and then
+  // will be passed as arguments to our component.
+  {
+    fetchCourses: catalogFetchCourses,
+  },
+)(injectIntl(CatalogPageBase));
```

Comment:
The `connect` based approach has somewhat fallen out of favour with hooks. 
What you can do in the component now is:

```javascript
const dispatch = useDispatch();
const courses = useSelector(selectCourses);

useEffect(() => {
  diapatch(catalogFetchCourses());
}, []);
```